### PR TITLE
Custom environment Management

### PR DESCRIPTION
**Pull request by albi3ro**
To properly install both the plugins and qchem from source into the environment, we can create our own environment, and then just link the python version at run time

```
asv run -E'existing:.asv/env/customenv/bin/python'
```

The script `customenv_build.sh` initializes the environment by:
*  cloning the plugins into `.asv/sources`
* creating a conda environment at `.asv/env/customenv`
* cloning pennylane into `.asv/env/customenv/project`, the location that asv would also store the source files for PennyLane
* performing editable installs on the plugins and qchem
* installing all dependencies

The script `update_sources.sh` will perform `git pull` on each plugin to update them.  

`customenv_build.sh` should be run once to create the environment, and `update_sources.sh` should be performed before each session if changes in the plugins might have occured.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -18,6 +18,10 @@ Check out the current results at [https://pennylaneai.github.io/benchmark/](http
 
 `benchmarks/benchmark_functions`: folder holding the basic benchmark functions which can be used independently of ASV.
 
+`customenv_build.sh`: clones plugin source code into `.asv/sources`, creates a conda environment in `.asv/env/customenv`, populates the custom environment with necessary packages
```
Comment (mariaschuld) [inline README.md]:
```suggestion
`customenv_build.sh`: clones plugin source code into `.asv/sources`, creates a conda environment in `.asv/env/customenv`, populates the custom environment with necessary packages.
```

File: customenv_build.sh [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+yes | .asv/env/customenv/bin/python -m pip install tensorflow
+yes | .asv/env/customenv/bin/python -m pip install torch
+yes | .asv/env/customenv/bin/python -m pip install networkx
+yes | .asv/env/customenv/bin/python -m pip install Qulacs
+yes | .asv/env/customenv/bin/python -m pip install qsimcirq
+
+
+git clone https://github.com/PennyLaneAI/pennylane.git .asv/env/customenv/project
+yes | .asv/env/customenv/bin/python -m pip install -e .asv/env/customenv/project/qchem
```
Comment (mariaschuld) [inline customenv_build.sh]:
```suggestion
yes | .asv/env/customenv/bin/python -m pip install -e .asv/env/customenv/project/qchem

```

File: update_sources.sh [hunk 3]
```diff
@@ -0,0 +1,9 @@
+cd .asv/sources/pennylane-lightning
+git pull
+cd ../pennylane-qiskit
+git pull
+cd ../pennylane-cirq
+git pull
+cd ../pennylane-qulacs
+git pull
+cd ../../..
```
Comment (mariaschuld) [inline update_sources.sh]:
```suggestion
cd ../../..

```

**Review by mariaschuld — APPROVED** [review]
I don't know much about this workflow, but happy if you found a solution.

Will this allow us to run the device benchmarks online too?

**Comment by albi3ro** [comment]
Locally last night I was able to execute all the tests but 'cirq.pasqal' and two of the 'default.mixed' were timing out. This solution should work fine on aws.

**Pull request merged**